pycyphal/dsdl/_import_hook.py

```diff
@@ -0,0 +1,136 @@
+
+
+class DsdlMetaFinder(MetaPathFinder):
+    def __init__(
+        self,
+        lookup_directories: Iterable[_AnyPath],
+        output_directory: _AnyPath,
+        allow_unregulated_fixed_port_id: bool,
+    ) -> None:
+        super().__init__()
+
+        _logger.debug("lookup dirs: %s", lookup_directories)
+        _logger.debug("output dir: %s", output_directory)
+
+        self.lookup_directories = list(map(str, lookup_directories))
+        self.output_directory = output_directory
+        self.allow_unregulated_fixed_port_id = allow_unregulated_fixed_port_id
+        self.root_namespace_directories: Iterable[pathlib.Path] = list()
+
+        # Build a list of root namespace directories from lookup directories.
+        # Any dir inside any of the lookup directories is considered a root namespace.
+        for dir in self.lookup_directories:
+            for namespace in pathlib.Path(dir).iterdir():
+                if namespace.is_dir():
+                    _logger.debug("Using root namespace %s at %s", namespace.name, namespace)
+                    self.root_namespace_directories.append(namespace)
+
+    def find_source_dir(self, root_namespace: str) -> Optional[pathlib.Path]:
+        """
+        Finds DSDL source directory for a given root namespace name.
+        """
+        for namespace_dir in self.root_namespace_directories:
+            if namespace_dir.name == root_namespace:
+                return namespace_dir
+        return None
+
+    def is_compiled(self, root_namespace: str) -> bool:
+        """
+        Returns true if given root namespace exists in output directory (compiled).
+        """
+        return pathlib.Path(self.output_directory, root_namespace).exists()
+
+    def find_spec(self, fullname, path, target=None):
+        _logger.debug("Attempting to load module %s as DSDL", fullname)
+
+        # DSDL root namespace is equal to module name
+        root_namespace = fullname
+
+        root_namespace_dir = self.find_source_dir(root_namespace)
+        if not root_namespace_dir:
+            return None
+
+        _logger.debug("Found root namespace %s in DSDL source directory %s", root_namespace, root_namespace_dir)
+
+        if not self.is_compiled(root_namespace):
+            _logger.debug("Compiling DSDL in %s", root_namespace_dir)
+            compile(
+                root_namespace_dir,
+                self.root_namespace_directories,
+                self.output_directory,
+                self.allow_unregulated_fixed_port_id,
+            )
+
+        compiled_module_dir = pathlib.Path(self.output_directory, root_namespace)
+        module_location = compiled_module_dir.joinpath("__init__.py")
+        submodule_locations = [str(compiled_module_dir)]
+
+        return spec_from_file_location(fullname, module_location, submodule_search_locations=submodule_locations)
+
+
+def get_default_lookup_dirs():
+    return os.environ.get("CYPHAL_PATH", "").split(os.pathsep)
```

Comment:
This will fail on Windows because you're using `:` in the orc file but it would expect `;`. Maybe we should normalize the value first?

Reply:
Oh yeah, I was wondering how to handle this. Ideally env vars should have OS-specific path separator, but it's hard to do for static configuration files like orc. We could split the path by both `:` and `;` if there are no other OS-specific path separators?

Reply:
How about this:

```suggestion
    return os.environ.get("CYPHAL_PATH", "").replace(os.pathsep, ";").split(";")
```